docs/fud/axi-gen.md

```diff
@@ -0,0 +1,72 @@
+# Axi generation
+
+Calyx currently generates a fairly complex AXI interface that can be daunting
+to deal with if confronting for the first time.
+The following is an overview of how the generation occurs
+and how the output AXI interface behaves as of 2022-9-11.
+
+In general, when `fud` is asked to create an [`.xclbin` file][xclbin] a `kernel.xml`,
+`main.sv`, and `toplevel.v` are created as intermediate steps for [xilinx tools][xilinx_tools]
+to properly work.
+
+`main.sv` contains the SystemVerilog needed for our computations to perform
+correctly. It is implemented as an FSM that derives from the original Calyx program.
+These files are very hard to read and not really conducive to being examined directly.
+[`kernel.xml`][kernel_xml] defines register maps and ports of our
+toplevel xilinx tools needs. `toplevel.v` wraps our computation kernel and contains
+the AXI interface for each memory define in a calyx program. 
+
+## Toplevel
+
+Our toplevel is generated through files in `src/backend/xilinx/`.
+
+### AXI memory controller
+Here, a calyx program is queried and all of its sub components are extracted.
+Most importantly, memories are extracted and separate AXI interfaces are created for each
+one (meaning, there is no shared bus between memories). Each memory has its own
+(single port) bram which writes data taken from an `mi_axi_RDATA` wire where i is the index of
+the memory. Eventually the brams are read and fed into the
+computation kernel of `main.sv`, which outputs results directly into the relevant
+memories as defined in the original Calyx program.
+
+For each memory we dynamically determine necesarry address and data widths and sizes.
```

Comment:
Not quite sure what you mean by "dynamically" here. Maybe just say that the pass looks at the cell declaration to determine its size?